src/neo/Oracle/Protocols/Https/OracleHttpsRequest.cs

```diff
@@ -0,0 +1,55 @@
+using Neo.IO;
+using System;
+using System.IO;
+using System.Text;
+
+namespace Neo.Oracle.Protocols.Https
+{
+    public class OracleHttpsRequest : OracleRequest
+    {
+        /// <summary>
+        /// HTTP Methods
+        /// </summary>
+        public HttpMethod Method { get; set; }
+
+        /// <summary>
+        /// Type
+        /// </summary>
+        public override OracleRequestType Type => OracleRequestType.HTTPS;
+
+        /// <summary>
+        /// URL
+        /// </summary>
+        public Uri URL { get; set; }
+
+        /// <summary>
+        /// Filter
+        /// </summary>
+        public string Filter { get; set; }
+
+        /// <summary>
+        /// Body
+        /// </summary>
+        public byte[] Body { get; set; }
+
+        /// <summary>
+        /// Get hash data
+        /// </summary>
+        /// <returns>Hash data</returns>
+        protected override byte[] GetHashData()
+        {
+            using (var stream = new MemoryStream())
+            using (var writer = new BinaryWriter(stream, Encoding.UTF8, true))
+            {
+                writer.Write((byte)Type);
+                writer.Write((byte)Method);
+                writer.WriteVarString(URL.ToString());
+                writer.WriteVarString(Filter);
+                if (Body != null) writer.WriteVarBytes(Body);
```

Comment:
If `Body` is `null`, should we write the length `0`? Or can it be `null`?

Reply:
With GET always will be null. If in the future we have POST, it will have content.

Reply:
If you prefer, we can remove it now, because there are no POST.

Reply:
Let's remove it now. We can add it back if we support `POST` in the future.